internal/auth/authn/jwtverifer.go

```diff
@@ -0,0 +1,54 @@
+package authn
+
+import (
+	"context"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+	"net/http"
+	"strings"
+)
+
+type tokenKeyType string
+
+const (
+	AuthHeaderKey              = "Authorization"
+	BearerScheme               = "Bearer "
+	tokenKeyValue tokenKeyType = "token"
+)
+
+type JWTAuth struct {
+	JwksUrl string
+}
+
+func (j JWTAuth) CheckPermission(ctx context.Context, resource string, op string) (bool, error) {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (j JWTAuth) AuthHandler(next http.Handler) http.Handler {
```

Comment:
Isn't this essentially the same as `K8sAuthN` ? Should it replace the the original implementation ?

Reply:
I think keep it separate. Retrieving JWKS certs from a kubernetes cluster can be tricky compared to an external SSO provider.